### PR TITLE
WD : Feat wishlist - Completed Wishlist Feature

### PR DESCRIPTION
**Pull request by devananth**
### File to be checked : 

**Components :**
      - Product 
      - Navbar
      - ProductsContainer
     
**Contexts :**
      - wishlist-context

**Pages :**
      - wishlist

**Reducers :**
      - wishlistReducer
      
 *Note : No Authentication is added , wishlist items are available only till the session expires*
 
 **Live URL:** https://deploy-preview-4--stoic-kowalevski-6ec98e.netlify.app/

### CODE REVIEW

File: src/components/Card/ProductCard/Product.jsx [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
+    rating,
+  } = productDetails;
+
+  const { wishlistState, wishlistDispatch } = useWishlist();
+
+  const { wishlist } = wishlistState;
+
+  const wishlistClickHandler = () => {
+    console.log(itemNotInList(_id, wishlist));
+
```
Comment (prathmeshjagtap) [inline src/components/Card/ProductCard/Product.jsx]: remove console.log

File: src/reducers/wishlistReducer.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { WISHLIST_ACTIONS } from "../Utils";
+
+const wishlistReducer = (state, action) => {
+  const { type, payload } = action;
+
+  const { wishlist } = state;
+
+  console.log(type, payload, wishlist);
+
```
Comment (prathmeshjagtap) [inline src/reducers/wishlistReducer.js]: you can remove the console.log

**Review by prathmeshjagtap — APPROVED** [review]
Code is Well structured
I liked you have made separate files for Constants 
and Utilities functions

File: src/components/Card/ProductCard/Product.jsx (hunk 1 above)
Comment (devananth) [reply src/components/Card/ProductCard/Product.jsx]: Thanks for pointing it out.

File: src/Utils/helperFunctions.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const itemNotInList = (id, arr) =>
+  arr.some(({ _id }) => id === _id) === false;
```
Comment (ashishnnnnn) [inline src/Utils/helperFunctions.js:2]: Liked the way of using utils functions for these checks on list..

**Review by ashishnnnnn — APPROVED** [review]

**Pull request merged**